raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java

```diff
@@ -0,0 +1,510 @@
+        when(deviceIdGetter.getDeviceId()).thenReturn(generateRandomString());
+        //act
+        presenter.processTransaction(data, ravePayInitializer);
+        //assert
+        verify(networkRequest)
+                .chargeCard(any(ChargeRequestBody.class),
+                        any(Callbacks.OnChargeRequestComplete.class));
+    }
+
+    @Test
+    public void getFee_networkError_showFetchFeeFailedCalled() {
+
+        presenter.fetchFee(generatePayload(), generateRandomInt());
+
+        ArgumentCaptor<Callbacks.OnGetFeeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnGetFeeRequestComplete.class);
+        verify(networkRequest).getFee(any(FeeCheckRequestBody.class), captor.capture());
+
+        captor.getAllValues().get(0).onError(generateRandomString());
+
+        verify(view).showFetchFeeFailed(anyString());
+
+    }
+
+    @Test
+    public void getFee_networkError_showFetchFeeFailedSuccessful() {
+
+        presenter.fetchFee(generatePayload(), generateRandomInt());
+
+        ArgumentCaptor<Callbacks.OnGetFeeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnGetFeeRequestComplete.class);
+        verify(networkRequest).getFee(any(FeeCheckRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateFeeCheckResponse());
+
+        verify(view).displayFee(anyString(), any(Payload.class), anyInt());
+
+    }
+
+    @Test
+    public void chargeCard_onSuccess_onPinAuthModelSuggested_showonPinAuthModelSuggested() {
+
+        presenter.chargeCard(generatePayload(), generateRandomString());
+        verify(view).showProgressIndicator(true);
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(PIN), generateRandomString());
+
+        verify(view).showProgressIndicator(false);
+        verify(view).onPinAuthModelSuggested(any(Payload.class));
+
+    }
+
+    @Test
+    public void chargeCard_onSuccess_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
+
+        presenter.chargeCard(generatePayload(), generateRandomString());
+        verify(view).showProgressIndicator(true);
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(AVS_VBVSECURECODE), generateRandomString());
+
+        verify(view).showProgressIndicator(false);
+        verify(view).onAVS_VBVSECURECODEModelSuggested(any(Payload.class));
+
+    }
+
+    @Test
+    public void chargeCard_onError_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
+
```

Comment:
@mikelis135 the name of your test does not correspond to the test you wrote here

Reply:
@mikelis135 and the name of the test does not make sense